viewing_party/party.py

```diff
@@ -1,23 +1,158 @@
+    if len(user_data["watched"]) > 0:
+        rating = []
+        for movie in user_data["watched"]:
+            rating.append(movie["rating"])
+        average_rating = sum(rating) / len(rating)
+        return average_rating
+    return 0
 
-
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) > 0:
+        janes_count = {}
+        for data in user_data["watched"]:
+            if data["genre"] not in janes_count:
+                janes_count[data["genre"]] = 1
+            else:
+                janes_count[data["genre"]] += 1
+        # popular_genre = max(janes_count, key = janes_count.get)
+        popular_genre = [key for key, value in janes_count.items() if value == max(janes_count.values())]
+        return ", ".join(popular_genre)
+    return None
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+
+    user_movies_list = []
+    friends_movies_list = []
+    unique_user_movies = []
+
+    for movie in user_data["watched"]:
+        user_movies_list.append(movie)
+
+    for data in user_data["friends"]:
+        for movie in data["watched"]:
+            if movie not in friends_movies_list:
+                friends_movies_list.append(movie)
+
+    for movie in user_data["watched"]:
+        if movie not in friends_movies_list:
+            unique_user_movies.append(movie)
+
+    return unique_user_movies
+
+def get_friends_unique_watched(user_data):
+
+    user_movies_list = []
+    friends_movies_list = []
+    unique_friend_movies = []
+
+    for movie in user_data["watched"]:
+        user_movies_list.append(movie)
+
+    for data in user_data["friends"]:
+        for movie in data["watched"]:
+            if movie not in friends_movies_list:
+                friends_movies_list.append(movie)
+
+    for movie in friends_movies_list:
+        if movie not in user_movies_list:
+            if movie not in unique_friend_movies:
+                unique_friend_movies.append(movie)
+
+    return unique_friend_movies
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    unique_friend_movies = get_friends_unique_watched(user_data)
+    recommendations = []
 
+    for movie in unique_friend_movies:
+        if movie["host"] in user_data["subscriptions"]:
+            recommendations.append(movie)
+
+    return recommendations
+    
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+
+    unique_friend_movies = get_friends_unique_watched(user_data)
+    user_genre_list = []
+    recommendations = []
+
+    for movie in user_data["watched"]:
+        user_genre_list.append(movie["genre"])
+
+    user_genre_dict = Counter(user_genre_list)

```

Comment:
Interesting application of `Counter`! Very cool making use of the `.most_common` method.